src/app/grid/grid-allData-summary/grid-allData-summary.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, OnInit, ViewChild } from "@angular/core";
+import {
+    IgxGridComponent,
+    IgxNumberSummaryOperand,
+    IgxSummaryResult } from "igniteui-angular";
+import { DATA } from "../../data/nwindData";
+
+class DiscountedSummary extends IgxNumberSummaryOperand {
+
+  constructor() {
+    super();
+  }
+
+  public operate(data?: any[], allData = [], fieldName = ""): IgxSummaryResult[] {
+    const result = [];
+    result.push({
+        key: "products",
+        label: "Producs",
+        summaryResult: data.length
+      });
+    result.push({
+        key: "total",
+        label: "Total Items",
+        summaryResult: IgxNumberSummaryOperand.sum(data)
+      });
+    result.push({
+        key: "discounted",
```

Comment:
Maybe you meant "Discontinued" instead of "discounted" ?

Reply:
I actually mean "Discounted" .. like reduced price ..